wger/utils/permissions.py

```diff
@@ -96,9 +96,4 @@ class CreateUserApiPermission(permissions.BasePermission):
     '''
 
     def has_permission(self, request, view):
-        user = request.user
-        if user == AnonymousUser():
-            return False
-        elif user.username == 'admin':
-            return True
-        return user.userprofile.create_user_via_api
+        return bool(request.user and request.user.is_authenticated)
```

Comment:
Kindly clarify here. Why do you have it taking in view as a parameter since I can't see it used?

Reply:
in the settings_global.py 
permissions classes are set to -> ` 'DEFAULT_PERMISSION_CLASSES': ('wger.utils.permissions.WgerPermission', ),`
When we try to add default permissions from Django we get errors. Basically, we over-rode the default IsAuthenticated class which takes in the request and the view as parameters to enable protection of the selected endpoint in the view.

Reply:
Thanks for the clarification